openedx/core/djangoapps/programs/utils.py

```diff
@@ -326,3 +333,32 @@ def _attach_course_run_upgrade_url(self, run_mode):
                 run_mode['upgrade_url'] = None
         else:
             run_mode['upgrade_url'] = None
+
+    def _attach_instructors(self):
+        """
+        Extend the program data with instructor data. The instructor data added here is persisted
+        on each course in modulestore and can be edited in Studio. Once the course metadata publisher tool
+        supports the authoring of course instructor data, we will be able to migrate course
+        instructor data into the catalog, retrieve it via the catalog API, and remove this code.
+        """
+        cache_key = 'program.instructors.{uuid}'.format(
+            uuid=self.data.get('uuid')
+        )
+        program_instructors = cache.get(cache_key)
+        if not program_instructors:
+            instructors_by_name = {}
+            module_store = modulestore()
+            for course in self.data['courses']:
+                for course_run in course['course_runs']:
+                    course_run_key = CourseKey.from_string(course_run['key'])
+                    course_descriptor = module_store.get_course(course_run_key)
+                    if course_descriptor:
+                        course_instructors = getattr(course_descriptor, 'instructor_info', {})
+                        # Deduplicate program instructors using instructor name
+                        instructors_by_name.update({instructor.get('name'): instructor for instructor
+                                                    in course_instructors.get('instructors', [])})
+
+            program_instructors = instructors_by_name.values()
+            cache.set(cache_key, program_instructors)
```

Comment:
@douglashall cool, I thought the default was `None`. Instructors likely don't change too often, so consider using at least an hour TTL; 5 minutes doesn't seem all that useful.